src/riot-rs-threads/src/lib.rs

```diff
@@ -118,20 +118,16 @@ impl Threads {
         arg: usize,
         stack: &'static mut [u8],
         prio: RunqueueId,
-    ) -> Option<&mut Thread> {
-        if let Some((thread, pid)) = self.get_unused() {
-            Cpu::setup_stack(thread, stack, func, arg);
-            thread.prio = prio;
-            thread.pid = pid;
-            thread.state = ThreadState::Paused;
-
-            Some(thread)
-        } else {
-            None
-        }
+    ) -> Option<ThreadId> {
+        let (thread, pid) = self.get_unused()?;
+        Cpu::setup_stack(thread, stack, func, arg);
+        thread.prio = prio;
+        thread.pid = pid;
+        thread.state = ThreadState::Paused;
+        Some(pid)
```

Comment:
Drive-by improvement because I noticed that the caller doesn't need the `&mut Thread`.